src/target/renesas.c

```diff
@@ -637,47 +798,16 @@ static void renesas_add_flash(target_s *t, target_addr_t addr, size_t length)
 	if (!priv_storage)
 		return;
 
-	/* Renesas RA MCUs can have one of two kinds of flash memory, MF3/4 and RV40
-	 * Flash type by series:
-	 * ra2l1 - MF4
-	 * ra2e1 - MF4
-	 * ra2e2 - MF4
-	 * ra2a1 - MF3
-	 * ra4m1 - MF3
-	 * ra4m2 - RV40
-	 * ra4m3 - RV40
-	 * ra4e1 - RV40
-	 * ra4w1 - MF3
-	 * ra6m1 - RV40
-	 * ra6m2 - RV40
-	 * ra6m3 - RV40
-	 * ra6m4 - RV40
-	 * ra6m5 - RV40
-	 * ra6e1 - RV40
-	 * ra6t1 - RV40
-	 * ra6t2 - RV40
-	 */
+	switch (priv_storage->details.flash_type) {
+	case FLASH_MF3:
+		DEBUG_WARN("Found renesas chip with Flash type MF3, not implemented\n");
+		return;
 
-	switch (priv_storage->series) {
-	case PNR_SERIES_RA2L1:
-	case PNR_SERIES_RA2E1:
-	case PNR_SERIES_RA2E2:
-	case PNR_SERIES_RA2A1:
-	case PNR_SERIES_RA4M1:
-	case PNR_SERIES_RA4W1:
+	case FLASH_MF4:
+		DEBUG_WARN("Found renesas chip with Flash type MF4, not implemented\n");
```

Comment:
You can merge these two, AFAIR, the driver will be the same when time comes to implement it, and the extra debug info is not useful as we have already have the family, better so save memory and use 1 string for both